src/indicator.jl

```diff
@@ -0,0 +1,57 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file extends JuMP to indicator constraints. It is a good example of how
+# JuMP can be extended.
+
+function build_indicator_constraint(
+    _error::Function, variable::JuMP.AbstractVariableRef, constraint::JuMP.ScalarConstraint, ::Type{MOI.IndicatorSet{A}}) where A
+
+    set = MOI.IndicatorSet{A}(moi_set(constraint))
+    return VectorConstraint([variable, jump_function(constraint)], set)
+end
+function _indicator_variable_set(::Function, variable::Symbol)
+    return variable, MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE}
+end
+function _indicator_variable_set(_error::Function, expr::Expr)
+    if expr.args[1] == :¬ || expr.args[1] == :!
+        if length(expr.args) != 2
+            _error("Invalid binary variable expression `$(expr)` for indicator constraint.")
+        end
+        return expr.args[2], MOI.IndicatorSet{MOI.ACTIVATE_ON_ZERO}
+    else
+        return expr, MOI.IndicatorSet{MOI.ACTIVATE_ON_ONE}
+    end
+end
+function parse_one_operator_constraint(
+    _error::Function, vectorized::Bool, ::Union{Val{:(=>)}, Val{:⇒}}, lhs, rhs)
+
+    variable, S = _indicator_variable_set(_error, lhs)
+    if !(rhs isa Expr) || rhs.head != :braces || length(rhs.args) != 1
```

Comment:
`!(rhs isa Expr) || rhs.head != :braces` can be written as `!isexpr(rhs, :braces)`. (`Meta.isexpr`)